### PR TITLE
Add auto-deploy to CloudRun regions

### PR DESCRIPTION
**Pull request by BobyMCbobs**
Related: https://github.com/kubernetes/k8s.io/issues/3577

### CODE REVIEW

**Comment by BobyMCbobs** [comment]
What might need to be done
- cloudrun admin/update permissions for the k8s-infra build service account
- postsubmit job?

**Comment by dims** [comment]
/approve

leaving LGTM for @ameukam 

@BobyMCbobs tested locally right?

**Comment by BobyMCbobs** [comment]
> /approve

Thank you @dims 

> leaving LGTM for @ameukam
> 
> @BobyMCbobs tested locally right?

I ran this against k8s-infra-ii-sandbox with Run Services brought up and with updated permissions for the build GCP service account

**Comment by dims** [comment]
thanks for confirming @BobyMCbobs

File: hack/make-rules/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+IMAGE_REPO="${IMAGE_REPO:-gcr.io/k8s-staging-infra-tools/archeio}"
+PROJECT="${PROJECT:-k8s-infra-oci-proxy}"
+
+REGIONS=(
+    us-central1
+    us-west1
+)
+
+# shellcheck disable=SC2048
+for REGION in ${REGIONS[*]}; do
```
Comment (ameukam) [inline hack/make-rules/deploy.sh:33]: Why not ? 
```suggestion
for REGION in "${REGIONS[*]}"; do
```

File: hack/make-rules/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    gcloud --project="${PROJECT}" \
```
Comment (ameukam) [inline hack/make-rules/deploy.sh:34]: Can be done later:
```suggestion
    gcloud beta --project="${PROJECT}" \
```

We can get access to some features usable and test for the sandbox environment.

File: cloudbuild.yaml [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+  - name: gcr.io/cloud-builders/docker
+    entrypoint: "/usr/bin/make"
+    args:
+      - "push-images"
+      - "TAG=$_GIT_TAG"
+  - name: "gcr.io/google.com/cloudsdktool/cloud-sdk"
+    entrypoint: "/usr/bin/make"
+    args:
+      - "deploy"
+      - "TAG=$_GIT_TAG"
```
Comment (ameukam) [inline cloudbuild.yaml:12]: We could introduce an order of steps to make sure the images are built before deployment. (Also we may have separate cloudbuild files). 
Ben's feedback can be really useful.

File: hack/make-rules/deploy.sh (hunk 1 above)
Comment (BobyMCbobs) [reply hack/make-rules/deploy.sh:33]: This won't iterate over all the values in $REGIONS, but will treat all values as a single string

File: hack/make-rules/deploy.sh (hunk 2 above)
Comment (BobyMCbobs) [reply hack/make-rules/deploy.sh:34]: Which features will this enable that we need?

File: cloudbuild.yaml (hunk 3 above)
Comment (BobyMCbobs) [reply cloudbuild.yaml:12]: I'm not sure what needs to be changed, currently it does build and push the image before deploying it to the staging environments

File: hack/make-rules/deploy.sh (hunk 2 above)
Comment (ameukam) [reply hack/make-rules/deploy.sh:34]: No features needed at the moment but might be useful in the features. Feel free to ignore my comment.

File: cloudbuild.yaml (hunk 3 above)
Comment (ameukam) [reply cloudbuild.yaml:12]: Something like that https://cloud.google.com/build/docs/configuring-builds/configure-build-step-order#examples. 

Feel free to ignore my comment if it's not needed.

Comment (BobyMCbobs) [reply cloudbuild.yaml]: My experience of using CloudBuild is that the jobs run consecutive, though it'll be pretty trivial to add this to ensure that always run in order.

Comment (BobyMCbobs) [reply cloudbuild.yaml]: Added in 1012a65

**Comment by BobyMCbobs** [comment]
Woohoo! Thank you @dims merging!

**Pull request merged**